perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_group_resource_attribute_def_def_o365EmailAddresses_mu.java

```diff
@@ -0,0 +1,120 @@
+ *     <li>no duplicates among the list values</li>
+ *     <li>no duplicates among all values of this attribute and all values of
+ *     attribute urn:perun:member:attribute-def:def:o365EmailAddresses:mu</li>
+ * </ul>
+ *
+ * @author Martin Kuba &lt;makub@ics.muni.cz>
+ */
+@SuppressWarnings("unused")
+public class urn_perun_group_resource_attribute_def_def_o365EmailAddresses_mu extends ResourceGroupAttributesModuleAbstract implements ResourceGroupAttributesModuleImplApi {
+
+	private static final String NAMESPACE = AttributesManager.NS_GROUP_RESOURCE_ATTR_DEF;
+	static final String ADNAME_ATTRIBUTE = NAMESPACE + ":adName";
+
+	@Override
+	public void checkAttributeValue(PerunSessionImpl sess, Resource resource, Group group, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
+		List<String> emails;
+		//get values
+		Object value = attribute.getValue();
+		if (value == null) {
+			throw new WrongAttributeValueException(attribute, resource, group, "can't be null.");
+		} else if (!(value instanceof List)) {
+			throw new WrongAttributeValueException(attribute, resource, group, "is of type " + value.getClass() + ", but should be ArrayList");
+		} else {
+			//noinspection unchecked
+			emails = (List<String>) value;
+		}
+		//check syntax of all values
+		for (String email : emails) {
+			Matcher emailMatcher = emailPattern.matcher(email);
+			if (!emailMatcher.matches())
+				throw new WrongAttributeValueException(attribute, resource, group, "Email " + email + " is not in correct form.");
+		}
+
+		//at least one value if adName is set
+		AttributesManagerBl am = sess.getPerunBl().getAttributesManagerBl();
+		try {
+			Object adName = am.getAttribute(sess, resource, group, ADNAME_ATTRIBUTE).getValue();
+			if (adName != null && emails.isEmpty()) {
+				throw new WrongAttributeValueException(attribute, resource, group, "at least one email must be defined");
+			}
+		} catch (AttributeNotExistsException | GroupResourceMismatchException e) {
+			throw new InternalErrorException(e.getMessage(), e);
+		}
+
+		//check for duplicities
+		if (hasDuplicate(emails)) {
+			throw new WrongAttributeValueException(attribute, resource, group, "duplicate values");
+		}
+
+		//check for duplicities among all members and groups
+		//TODO not implemented yet
+	}
+
+	/**
+	 * Prefills value created by joining value of urn:perun:group_resource:attribute-def:def:adName with "@group.muni.cz"
+	 */
+	@Override
+	public Attribute fillAttribute(PerunSessionImpl sess, Resource resource, Group group, AttributeDefinition attrDef) throws InternalErrorException, WrongAttributeAssignmentException {
+		if (!NAMESPACE.equals(attrDef.getNamespace())) throw new WrongAttributeAssignmentException(attrDef);
+		try {
+			Attribute result = new Attribute(attrDef);
+			Object adName =
```

Comment:
Not sure, why splitted to 2 lines ?

Reply:
just to make it shorter, but can be put on one line.